### PR TITLE
Fix missing docstrings, improve test coverage, and improve error mess…

### PR DESCRIPTION
**Pull request by drevell**
…age for nonexistent mysql version

### CODE REVIEW

File: mysqltest/config.go [hunk 1]
```diff
@@ -68,6 +72,15 @@ func WithVersion(v string) Option {
 	}
 }
 
+// WithLogger overrides the default logger. This logger will receive messages about MySQL startup
+// progress. The default is to use the go "log" package.
+func WithLogger(l Logger) Option {
```
Comment (sethvargo) [inline mysqltest/config.go:77]: No action, but you should check out jba's slog proposal on the Go repo if you're interested for structured logging in the stdlib.

File: mysqltest/mysqltest_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	if conf.mySQLVersion != "2" {
+		t.Errorf(`got mySQLVersion=%v", want "2"`, conf.mySQLVersion)
+	}
+	if _, ok := conf.progressLogger.(*fakeLogger); !ok {
+		t.Errorf("got progressLogger type %T, want %T", conf.progressLogger, logger)
+	}
+}
+
+type fakeLogger struct{}
```
Comment (sethvargo) [inline mysqltest/mysqltest_test.go:64]: For testing, we might want to create a `testLogger` that calls `t.Log`. That way we see the log output on test failures:

```go
type testLogger struct {
  tb testing.TB
}

func (l *testLogger) Printf(msg string, args ...any) {
  l.tb.Logf(msg, args...)
}
```

Then &testLogger{t} should just work.

**Review by sethvargo — APPROVED** [review]

**Review by yolocs — APPROVED** [review]
Side feature request: I wonder if it's reasonable to allow images not on docker registry (e.g. GCR mirror). E.g. GCR/AR probably have better reliability.

File: mysqltest/config.go (hunk 1 above)
Comment (drevell) [reply mysqltest/config.go:77]: Thanks, I hadn't seen that but I welcome it. https://github.com/golang/go/discussions/54763 for posterity.

File: mysqltest/mysqltest_test.go (hunk 2 above)
Comment (drevell) [reply mysqltest/mysqltest_test.go]: Done, thanks for the idea.

**Comment by drevell** [comment]
> Side feature request: I wonder if it's reasonable to allow images not on docker registry (e.g. GCR mirror). E.g. GCR/AR probably have better reliability.

Hmm good point, I hadn't given much thought to reliability of docker image downloads.

**Pull request merged**